### PR TITLE
changed flax_network call method to operate on whole epoch data.

### PR DESCRIPTION
**Pull request by Dalouvid**
Fix #

#### Summary of additions and changes

* the call method of a flax newtwork now uses a vectorized version of the apply_fn. It can operate on an array of feature data to compute logits/values for a whole episode
* in the loss, the vectroized versions are called

#### How to test this pull request

```

```

### CODE REVIEW

**Review by SamTov — APPROVED** [review]
Looks great! Thanks for the quick changes
